featuretools/entityset/entityset.py

```diff
@@ -163,77 +156,99 @@ def is_metadata(self):
-                'time_index': e.time_index,
-                'secondary_time_index': e.secondary_time_index,
-                'variables': {
-                    v.id: v.create_metadata_dict()
-                    for v in e.variables
-                },
-                'has_last_time_index': e.last_time_index is not None
-            } for eid, e in self.entity_dict.items()},
-        }
+    def create_data_description(self):
+        '''Serialize entityset to data description.
+
+        Args:
+            es (EntitySet) : Instance of :class:`.EntitySet`.
+
+        Returns:
+            description (dict) : Description of :class:`.EntitySet`.
+        '''
+        entities = dict(map(serialization.to_entity_descr, self.entities))
+        relationships = list(map(serialization.to_relation_descr, self.relationships))
+        return {'id': self.id, 'entities': entities, 'relationships': relationships}
 
     @classmethod
-    def from_metadata(cls, metadata, data_root=None):
-        es = EntitySet(metadata['id'])
-        set_last_time_indexes = False
-        for eid, entity in metadata['entity_dict'].items():
-            df, variable_types = cls._load_dummy_entity_data_and_variable_types(entity)
-            if data_root is not None:
-                df = load_entity_data(entity, data_root)
-            es.entity_from_dataframe(eid,
-                                     df,
-                                     index=entity['index'],
-                                     time_index=entity['time_index'],
-                                     secondary_time_index=entity['secondary_time_index'],
-                                     variable_types=variable_types)
-            if entity['has_last_time_index']:
-                set_last_time_indexes = True
-            for vid, v in entity['variables'].items():
-                if v['interesting_values'] and len(v['interesting_values']):
-                    es[eid][vid].interesting_values = v['interesting_values']
-        for rel in metadata['relationships']:
-            es.add_relationship(Relationship(
-                es[rel['parent_entity']][rel['parent_variable']],
-                es[rel['child_entity']][rel['child_variable']],
-            ))
-        if set_last_time_indexes:
-            es.add_last_time_indexes()
+    def from_data_description(cls, descr, **kwargs):
+        '''Deserialize entityset from data description.
+
+        Args:
+            descr (dict) : Description of :class:`.EntitySet`.
+            kwargs (keywords): Additional keyword arguments to pass as keywords arguments to the underlying deserialization method.
+
+        Returns:
+            es (EntitySet) : Instance of :class:`.EntitySet`.
+        '''
+        es = cls(descr['id'])
+        root = descr.get('root')
+
+        lti = []
+        for id, e in descr['entities'].items():
+            e['loading_info']['params'].update(kwargs)
+            k = serialization.from_entity_descr(e)
```

Comment:
for this line and the next 3, aim for readability of variable names

Reply:
Done